File: Students consultations/StudentsConsultations.Domain/Konsultacije.cs
using StudentsConsultations.Data.Domain;
using System;

namespace StudentsConsultations.Data.Domain
{
    public class Konsultacije 
    {
        public int StudentId { get; set; }
        public int NastavnikId { get; set; }
        public DateTime DatumKonsultacija { get; set; }

        public int RazlogId { get; set; }

        public Datum DatumObjekat { get; set; }
        public Nastavnik Nastavnik { get; set; }
        public Razlog Razlog { get; set; }
        public Student Student { get; set; }
    }
}

File: Students consultations/StudentsConsultations.Web/src/app/models/konsultacije.ts
import { Student } from './student';
import { Nastavnik } from './nastavnik';
import { DateObject } from 'ngx-bootstrap/chronos/types';
import { Projekat } from './projekat';
import { Ispit } from './ispit';
import { ZavrsniRad } from './zavrsniRad';

export class Konsultacije {
  studentId: number;
  nastavnikId: number;
  razlogId: number;
  odrzane: boolean;
  datumKonsultacija: Date;
  datumString: string;

  student: Student;
  nastavnik: Nastavnik;
  razlog: Razlog;
}

export class Razlog {
  razlogId: number;
  opis: string;

  projekat: Projekat;
  ispit: Ispit;
  zavrsniRad: ZavrsniRad;

  type: RazlogType;
  nazivTipa: string;
  nazivIspita: string;
  tipZavrsnogRada: string;
}

export enum RazlogType {
  Ispit = 0,

  ZavrsniRad,

  Projekat
}
